sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/jdbc/connection/ConnectionProviderSuite.scala

```diff
@@ -45,6 +54,84 @@ class ConnectionProviderSuite extends ConnectionProviderSuiteBase with SharedSpa
     }
   }
 
+  test("Throw an error selecting from an empty list of providers on create") {
```

Comment:
Could you add some tests for the case where a user setting both `DISABLED_JDBC_CONN_PROVIDER_LIST` and `connectionProvider`?

Reply:
Added

Reply:
Unfortunately, `DISABLED_JDBC_CONN_PROVIDER_LIST` is a static conf and `providers` is not lazy, i.e. I am not sure how to reevaluate providers in ConnectionProvider once the class has been loaded without reflection.